File: selection_sort/selection_sort.rb

def selection_sort(input)
  for i in 0..input.length-1
    for j in i..input.length-1
      if input[i] > input[j]
        input[i], input[j] = input[j], input[i]
      end
    end
  end
  return input
end


rand = Random.new
input = Array.new(500) { |num| rand.rand(500) }

p selection_sort(input)

File: selection_sort/selection_sort.py
import random


def selection_sort(input):
    for i in range(len(input)):
        for j in range(i, len(input)):
            if input[i] > input[j]:
                input[i], input[j] = input[j], input[i]
    return input


if __name__ == '__main__':
    input_list = sorted(random.sample(range(1000), 50))
    print(selection_sort(input_list))